Source/r_sky.c

```diff
@@ -67,6 +71,73 @@ void R_InitSkyMap (void)
   skytexturemid = 100*FRACUNIT;
 }
 
+static byte V_GetPaletteIndex(byte *palette, int r, int g, int b)
+{
+  byte best;
+  int best_diff, diff;
+  int i;
+
+  best = 0; best_diff = INT_MAX;
+
+  for (i = 0; i < 256; ++i)
+  {
+    diff = (r - palette[3 * i + 0]) * (r - palette[3 * i + 0])
+          + (g - palette[3 * i + 1]) * (g - palette[3 * i + 1])
+          + (b - palette[3 * i + 2]) * (b - palette[3 * i + 2]);
+
+    if (diff < best_diff)
+    {
+      best = i;
+      best_diff = diff;
+    }
+
+    if (diff == 0)
+    {
+      break;
+    }
+  }
+
+  return best;
+}
+
+byte R_SkyBlendColor(int tex)
+{
+  byte *pal = W_CacheLumpName("PLAYPAL", PU_STATIC);
+  int i, r = 0, g = 0, b = 0;
+  byte ret;
+
+  const int width = texturewidth[tex];
+
+  static int prevtex = -1;
+  static byte prevret = -1;
+
+  if (tex == prevtex)
+  {
+    return prevret;
+  }
+
+  // [FG] count colors
+  for (i = 0; i < width; i++)
+  {
+    byte *c = R_GetColumn(tex, i);
+    r += pal[3 * c[0] + 0];
+    g += pal[3 * c[0] + 1];
+    b += pal[3 * c[0] + 2];
+  }
+
+  r /= width;
+  g /= width;
+  b /= width;
+
+  // Get 1/3 for empiric reasons
+  ret = V_GetPaletteIndex(pal, r/3, g/3, b/3);
```

Comment:
This returns a color that is very much darker than the average sky top color. Is this intended?

Reply:
Yes this is intended I copied it form EE. Darker color looks better, for example on Doom 2 map15.